packages/perseus/src/widgets/matcher/matcher.tsx

```diff
@@ -12,37 +12,58 @@ import Renderer from "../../renderer";
 import {getPromptJSON as _getPromptJSON} from "../../widget-ai-utils/matcher/matcher-ai-utils";
 
 import type {SortableOption} from "../../components/sortable";
-import type {WidgetExports, WidgetProps, Widget} from "../../types";
+import type {WidgetExports, Widget, UniversalWidgetProps} from "../../types";
 import type {MatcherPromptJSON} from "../../widget-ai-utils/matcher/matcher-ai-utils";
 import type {
     PerseusMatcherWidgetOptions,
     PerseusMatcherUserInput,
+    MatcherPublicWidgetOptions,
 } from "@khanacademy/perseus-core";
+import type {PropsFor} from "@khanacademy/wonder-blocks-core";
 
 const HACKY_CSS_CLASSNAME = "perseus-widget-matcher";
 
-type RenderProps = PerseusMatcherWidgetOptions;
+interface InternalProps extends UniversalWidgetProps {
+    left: string[];
+    right: string[];
+    labels: string[];
+    orderMatters: boolean;
+    padding: boolean;
+}
 
-type Props = WidgetProps<RenderProps>;
+type ExternalProps = PropsFor<typeof Matcher>;
 
-type DefaultProps = {
-    left: Props["left"];
-    right: Props["right"];
-    labels: Props["labels"];
-    orderMatters: Props["orderMatters"];
-    padding: Props["padding"];
-    problemNum: Props["problemNum"];
-    onChange: Props["onChange"];
-    linterContext: Props["linterContext"];
-};
+type RenderProps = Omit<ExternalProps, keyof UniversalWidgetProps>;
```

Comment:
I personally find this way of deriving RenderProps from ExternalProps which is derived from the component props to be a little backwards conceptually. To me it adds to the confusion rather than mitigates it, which I feel like is compounded by us doing it for some widgets and not all of them.

I think we should stick to what we've been doing until there's a consensus about what we want and then do a PR that changes all of them at once. Maybe a small, inter-LEMS Perseus ADR? (Outside the normal ADR process?) Even if the goal is just to better explain the why, so that it's not adding to the confusion.